drivers/timer/stm32_lptim_timer.c

```diff
@@ -61,11 +65,41 @@ static const struct device *const clk_ctrl = DEVICE_DT_GET(STM32_CLOCK_CONTROL_N
  *    0xFFFF / (LSE freq (32768Hz) / 128)
  */
 
-static uint32_t lptim_clock_freq = KHZ(32);
-static int32_t lptim_time_base;
+#if (CONFIG_STM32_LPTIM_CLOCK_LSI)
+static uint32_t lptim_clock_freq = KHZ(32); /* Force STM32_SRC_LSI */
```

Comment:
I don't know how we were handling this previously but on some series, LSI could be different from 32K
L0, L1, F0, F1, F3
Not all of them use LPTIM driver (Only L0 today), but a comment on this case would be welcome.

Reply:
L0 uses LPTIM

Reply:
> We could probably use `STM32_LSI_FREQ` defined in stm32_clock_control.h. That would take care of cases where LSI is not 32 kHz.

in case  the LSI is 37kHz like stm32l0x, the lptim_clock_freq  is set to 32kHz so this cannot be STM32_LSI_FREQ